projects/packages/my-jetpack/_inc/components/my-jetpack-tab-panel/index.tsx

```diff
@@ -15,6 +16,22 @@ import { getMyJetpackSections, isValidMyJetpackSection } from './utils';
 export function MyJetpackTabPanel() {
 	const params = useParams();
 	const navigate = useNavigate();
+	const { filteredUnownedProducts, isLoading } = useFilteredProducts();
+
+	const showProductsTab = useMemo( () => {
+		return filteredUnownedProducts.length > 0 && ! isLoading;
+	}, [ filteredUnownedProducts.length, isLoading ] );
+
+	const availableTabs = useMemo(
+		() => getMyJetpackSections( showProductsTab ),
+		[ showProductsTab ]
+	);
+
+	// If the tab is not valid, use the default one.
+	const initialTab = useMemo( () => {
+		const validTab = isValidMyJetpackSection( params.section, showProductsTab );
+		return validTab ? params.section : MY_JETPACK_SECTION_OVERVIEW;
+	}, [ params.section, showProductsTab ] );
```

Comment:
This doesn't need memoization as it's a scalar value and the computation is not very expensive. This can be reverted to the original with that extra param passed.

Reply:
> it will prevent executing the function between rerenders

Yes, that is obviously what `useMemo` is for, but here its use may be more expensive than its benefit, for the computation and result being pretty simple.

Reply:
Yeah thanks, I already approved the PR. Let us merge it.